### ISSUE TITLE
Webhook module sends empty payloads

### ISSUE THREAD

**Issue by DavidAnderson684** [freescout-helpdesk/freescout#3040]
PHP version: 8.1.19
FreeScout version: 1.8.77

We use the webhook module (current version: 1.0.57) to send a separate application notifications of convo.created events.

Our receiving application receives a lot of incoming events from this with no body, i.e. no payload.

We have inspected the Freescout code to see what is happening and have located the code that sends an empty payload, but have not identified the root source of the problem behind it.

In `Modules/ApiWebhooks/Entities/Webhook.php`, in the run() method, the same input is being made into that plugin every time that `freescout:webhooks-process` is run (which for us is every minute). Currently there are over 200 such events. For these problematic/repeating ones, the $data parameter to this method (2nd parameter) being fed in is a 24-member array. The method feeds this array (first parameter, `$entity`) to `ApiWebhooks::formatEntity()` which is in `Modules/ApiWebhooks/Providers/ApiWebhooksServiceProvider.php`. However, `formatEntity()` has no code that can process an array for that parameter. Consequently, it returns `null`.

This leads to the payload parameter back in `run()`, i.e. `$options['json']`, also being null.

Our receiving application still returns an HTTP 200 when it receives the invalid payload, but this still doesn't result in Freescout not sending the event again: it sends it again, every time. As I say, there are over 200 such events each time now.

We deleted some conversations associated with the problematic notifications, but new ones are always coming in, so we need a more permanent fix.

This can be reproduced on demand by calling `artisan freescout:webhooks-process`. No checking of the mailbox is involved.

Here is an example of array contents being received in `$data` (output with `print_r())`:

```
Array
(
    [id] => 46531
    [number] => 46531
    [threadsCount] => 0
    [type] => email
    [folderId] => 13
    [status] => active
    [state] => published
    [subject] => Example.Com Support Request
    [preview] => Support request from Example.ComSupport requestYour Name:Jorg LöckmannOrder number for your purchase:#123924What kind of support request is this?:I need ongoing usage support
    [mailboxId] => 13
    [assignee] => 
    [createdBy] => Array
        (
            [id] => 10437
            [type] => customer
            [firstName] => 
            [lastName] => 
            [photoUrl] => 
            [email] => user@example.com
        )

    [createdAt] => 2023-03-13T18:57:17Z
    [updatedAt] => 2023-03-13T18:57:17Z
    [closedBy] => 
    [closedByUser] => 
    [closedAt] => 
    [userUpdatedAt] => 
    [customerWaitingSince] => Array
        (
            [time] => 2023-03-13T18:57:17Z
            [friendly] => Just now
            [latestReplyFrom] => customer
        )

    [source] => Array
        (
            [type] => email
            [via] => customer
        )

    [cc] => Array
        (
        )

    [bcc] => Array
        (
        )

    [customer] => Array
        (
            [id] => 10437
            [type] => customer
            [firstName] => 
            [lastName] => 
            [photoUrl] => 
            [email] => user@example.com
        )

    [_embedded] => Array
        (
            [threads] => Array
                (
                    [0] => Array
                        (
                            [id] => 72040
                            [type] => customer
                            [status] => active
                            [state] => published
                            [action] => Array
                                (
                                    [type] => 
                                    [text] => user@example.com started a new conversation #46531
                                    [associatedEntities] => Array
                                        (
                                        )

                                )

                            [body] => <div class="rcmBody" style="-webkit-text-size-adjust: none;
margin: 0">
        (snip)

</div>
                            [source] => Array
                                (
                                    [type] => email
                                    [via] => customer
                                )

                            [customer] => Array
                                (
                                    [id] => 10437
                                    [type] => customer
                                    [firstName] => 
                                    [lastName] => 
                                    [photoUrl] => 
                                    [email] => user@example.com
                                )

                            [createdBy] => Array
                                (
                                    [id] => 10437
                                    [type] => customer
                                    [firstName] => 
                                    [lastName] => 
                                    [photoUrl] => 
                                    [email] => user@example.com
                                )

                            [assignedTo] => 
                            [to] => Array
                                (
                                    [0] => support@example.com
                                )

                            [cc] => Array
                                (
                                )

                            [bcc] => Array
                                (
                                )

                            [createdAt] => 2023-03-13T18:57:17Z
                            [openedAt] => 
                            [_embedded] => Array
                                (
                                    [attachments] => Array
                                        (
                                        )

                                )

                        )

                )

        )

)
```

As a temporary fix we've caused `run()` to `return;` if the data won't be processed, but as I say, the method will then keep being called hundreds of times as a result.

**Comment by freescout-helpdesk** [comment]
How are you creating the conversation? Step-by-step instruction would be helpful.

**Comment by DavidAnderson684** [comment]
All incoming tickets are received by email. Some of those emails are sent directly by the customer from their email client, but the great majority come from a support form on our website that generates an email to the Freescout-linked mailbox. The example above is one of the latter.

I should clarify that it's only a small minority that end up with this problem. There have been thousands of tickets processed in the timespan that the 200 problematic notification events have accumulated during.

Looking at the conversation for the example above, I can also see that a successful notification also occurred. i.e. The webhook was at some point sent to the remote URL with the expected/correct payload; this is known because the ticket has had a note added that is added by that application which receives the webhook. So apparently some conversations end that come into the webhook function with the correct data, as well as then also eternally repeating with an invalid parameter, the array.

**Comment by freescout-helpdesk** [comment]
Fixed in API & Webhooks Module v1.0.58

**Comment by DavidAnderson684** [comment]
I can confirm that the updated module has fixed the problem.
